### ISSUE TITLE
tritonserver exits with Signal (11) received when receiving an async_stream_infer request with an unmatched model_name

### ISSUE THREAD

**Issue by songkq** [triton-inference-server/server#6091]
The tritonserver crashes and exits  when receiving an `async_stream_infer` request with an unmatched `model_name`. Could you please give some advice for this issue?

```
I0722 12:01:17.732586 2435 stream_infer_handler.cc:122] New request handler for ModelStreamInferHandler, 0

Signal (11) received.
 0# 0x00005580D50C763D in /opt/tritonserver/bin/tritonserver
 1# 0x00007FF89D097520 in /lib/x86_64-linux-gnu/libc.so.6
 2# TRITONSERVER_InferenceRequestId in /opt/tritonserver/bin/../lib/libtritonserver.so
 3# 0x00005580D512DF11 in /opt/tritonserver/bin/tritonserver
 4# 0x00005580D511C7BC in /opt/tritonserver/bin/tritonserver
 5# 0x00007FF89D3592B3 in /lib/x86_64-linux-gnu/libstdc++.so.6
 6# 0x00007FF89D0E9B43 in /lib/x86_64-linux-gnu/libc.so.6
 7# clone in /lib/x86_64-linux-gnu/libc.so.6
```

**Comment by dyastremsky** [comment]
We have filed a ticket to look at this bug. It would expedite our efforts if you can provide steps for a reproduction. This would include:
- Triton version
- Whether you used a Triton container or built Triton on your own (if so, instructions for building)
- Sample models (you can exclude in this case if this bug occurs for any models across any backends)
- Commands used to run Triton

**Comment by rmccorm4** [comment]
@songkq @Wecombo does this only happen if you have `--log-verbose=1` enabled? If you disable it, do you still see this issue?

If so, this should already be [fixed](https://github.com/triton-inference-server/server/pull/5842/files) and I believe is included in the 23.07 release already.

**Comment by GuanLuo** [comment]
Closing due to the issue may have been fixed and waited on user verification for more than two weeks.
